src/infra/src/table/migration/m20241220_000500_delete_meta_users_table.rs

```diff
@@ -0,0 +1,169 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Deletes user records from the meta table.
+
+use std::collections::HashMap;
+
+use config::meta::user::{DBUser, UserOrg};
+use sea_orm::{
+    ColumnTrait, EntityTrait, PaginatorTrait, QueryFilter, QueryOrder, TransactionTrait,
+};
+use sea_orm_migration::prelude::*;
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        let txn = manager.get_connection().begin().await?;
+        meta::Entity::delete_many()
+            .filter(meta::Column::Module.eq("user"))
+            .exec(&txn)
+            .await?;
+        txn.commit().await?;
+        Ok(())
+    }
+
+    async fn down(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        // The deletion of records from the meta table is not reversable.
+        let db = manager.get_connection();
+        let txn = db.begin().await?;
+        // Migrate pages of 100 records at a time to avoid loading too many
+        // records into memory.
+        // txn.execute()
+        let mut db_users = HashMap::new();
+        let mut user_pages = users::Entity::find()
+            .order_by_asc(users::Column::Id)
+            .paginate(&txn, 100);
+
+        while let Some(users) = user_pages.fetch_and_next().await? {
+            for u in users {
+                let db_user = DBUser {
+                    email: u.email.clone(),
+                    first_name: u.first_name.clone(),
+                    last_name: u.last_name.clone(),
+                    password: u.password.clone(),
+                    salt: u.salt.clone(),
+                    password_ext: u.password_ext.clone(),
+                    is_external: u.user_type != 0,
+                    organizations: vec![],
+                };
+                db_users.insert(u.email.clone(), db_user);
+            }
+        }
+
+        let mut org_user_pages = org_users::Entity::find()
+            .order_by_asc(org_users::Column::Id)
+            .paginate(&txn, 100);
+
+        while let Some(org_users) = org_user_pages.fetch_and_next().await? {
+            for ou in org_users {
+                let db_user = db_users
+                    .get_mut(&ou.email)
+                    .ok_or_else(|| DbErr::Migration("User not found".to_string()))?;
+                db_user.organizations.push(UserOrg {
+                    name: ou.org_id.clone(),
+                    role: ou.role.into(),
+                    token: ou.token.clone(),
+                    rum_token: ou.rum_token.clone(),
+                });
+            }
+        }
```

Comment:
this `down` migration doesn't seem to be doing anything. Maybe it's not a concern, but if it's really needed, the data `db_suers` that's read from those two newly created table can be put back to the `meta` table. seems like that's the intend, but the actual db insertion is missing.

Reply:
Oops! missed it. But as discussed earlier, upgrading to this version will be irreversible, hence I am cleaning up this whole down implementation.